src/main/java/uk/gov/ons/census/fwmt/csvservice/service/impl/CSVConverterServiceImpl.java

```diff
@@ -25,57 +17,108 @@
-      }
+    @Value("${gcpBucket.celocation}")
+    private Resource cePath;
+
+    @Value("${gcpBucket.ccslocation}")
+    private Resource ccsPath;
+
+    @Value("${gcpBucket.celocation}")
+    private Path cePathFileName;
+
+    @Value("${gcpBucket.ccslocation}")
+    private Path ccsPathFileName;
+
+    @Autowired
+    private CSVAdapterService csvAdapterService;
+
+    @Autowired
+    private GatewayEventManager gatewayEventManager;
+
+    public void convertCSVToCanonical(String ingestType) throws GatewayException {
+
+        CsvToBean<CSVRecordDTO> csvToBean;
+        Date formattedDate = new Date(System.currentTimeMillis());
+        Path csvPath;
+        Resource csvGCPFile;
+        SimpleDateFormat formatDate = new SimpleDateFormat("yyyy-MM-dd'_'HH-mm-ss");
+        String csvToUpdate;
+        String dateAndTime;
+        String csvFileExtension;
+        String processCSVExtension;
+
+        if (ingestType.equals("CEIngest")) {
+            csvGCPFile = cePath;
+            csvPath = cePathFileName;
+        } else {
+            csvGCPFile = ccsPath;
+            csvPath = ccsPathFileName;
+        }
+
+        try {
+            csvToBean = new CsvToBeanBuilder(
+                    new InputStreamReader(csvGCPFile.getInputStream(), StandardCharsets.UTF_8))
+                    .withType(CSVRecordDTO.class)
+                    .build();
+
+            csvToUpdate = Files.readString(csvPath, StandardCharsets.UTF_8);
+
+            dateAndTime = formatDate.format(formattedDate);
+
+            csvFileExtension = csvGCPFile.getFilename();
+
+            if (csvFileExtension != null) {
+                processCSVExtension = csvFileExtension.replace(".csv", ".processed_" + dateAndTime);
+                try (OutputStream os = ((WritableResource) csvGCPFile.createRelative(processCSVExtension)).getOutputStream()) {
+                    os.write(csvToUpdate.getBytes(StandardCharsets.UTF_8));
+                }
+            }
+
+        } catch (IOException e) {
+            String msg = "Failed to convert CSV to Bean.";
+            gatewayEventManager.triggerErrorEvent(this.getClass(), msg, "N/A", GatewayEventsConfig.UNABLE_TO_READ_CSV);
+            throw new GatewayException(GatewayException.Fault.SYSTEM_ERROR, e, msg);
+        }
+
+        if (ingestType.equals(CSVConverterService.CE_INGEST)) {
+            for (CSVRecordDTO csvRecordDTO : csvToBean) {
+                csvAdapterService.sendJobRequest(CanonicalJobHelper.createCEJob(csvRecordDTO));
+                gatewayEventManager.triggerEvent(String.valueOf(csvRecordDTO.getCaseId()), CSV_CE_REQUEST_EXTRACTED);
+            }
+
+            try {
+                Files.delete(csvPath);
+            } catch (IOException e) {
+                throw new GatewayException(GatewayException.Fault.SYSTEM_ERROR, e, "Unable to delete CE Ingest CSV");
+            }
+
+        } else {
+            for (CSVRecordDTO csvRecordDTO : csvToBean) {
```

Comment:
could potentially use an interface and have an implementation for CE and CCS - would get rid of the need for the if statement

Reply:
I agree, cuz we are going to need to ingest 2 more csv types and it will start to get complicated.
Also it looks like CSVRecordDTO is used for both CE and CCS, we should have separate ones.
See what happens if one of the rows fail, does it carry on? Should it?